rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -299,15 +300,12 @@ public static FlowNode getFlowEndNode(FlowExecution execution) {
     public static List<FlowNode> getChildNodes(final FlowNode parentNode) {
         final List<FlowNode> nodes = new ArrayList<FlowNode>();
 
-        FlowNodeNavigator nodeNavigator = new FlowNodeNavigator(new FlowNodeNavigationListener() {
-            @Override
-            public void onNode(FlowNode navNode) {
-                if (navNode.getParents().contains(parentNode) && !nodes.contains(navNode)) {
-                    nodes.add(navNode);
-                }
+        FlowGraphWalker walker = new FlowGraphWalker(parentNode.getExecution());
+        for (FlowNode node : walker) {
+            if (node.getParents().contains(parentNode) && !nodes.contains(node)) {
+                nodes.add(node);
```

Comment:
> I wasn't going to assume anything wrt hashing/equals impls

[the impls](https://github.com/jenkinsci/workflow-plugin/blob/82e7defa37c05c5f004f1ba01c93df61ea7868a5/api/src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java#L315-L327)